docs/general/developer-resources/emerald-paratime/writing-dapps-on-emerald.md

```diff
@@ -4,39 +4,173 @@ description: How to write and deploy a dApp to Oasis Emerald
+## Running Private Oasis Network Locally
+
+For convenient development and testing of your dApps the Oasis team prepared a
+docker image which brings you a complete Oasis stack to your desktop. This
+network is isolated from the Mainnet or Testnet and consists of:
+- one Oasis validator node with 1 second block time,
+- three Oasis compute nodes running Emerald ParaTime,
+- Emerald Web3 gateway with transaction indexer,
+- a helper script which populates initial test accounts for you.
+
+To run the image, type:
+
+```sh
+docker run -it -p8545:8545 -p8546:8546 oasisprotocol/emerald-dev
+```
+
+After a while, the tool will show you something like this:
+
+```
+emerald-dev 2022-03-11-git38da200 (oasis-core: 22.0, emerald-paratime: 7.1.0-rc1, emerald-web3-gateway: 2.0.0-rc1-git38da200)
+
+Starting oasis-net-runner with Emerald ParaTime...
+Starting postgresql...
+Starting emerald-web3-gateway...
+Populating account(s) (this might take a moment)...
+
+Available Accounts
+==================
+(0) 0x75eCF0d4496C2f10e4e9aF3D4d174576Ee9010E2 (100 ROSE)
+(1) 0x903a7dce5a26a3f4DE2d157606c2191740Bc4BC9 (100 ROSE)
+(2) 0xF149ad5CBFfD92ba84F5784106f6Cb071A32a1b8 (100 ROSE)
+(3) 0x2315F40C1122400Df55483743B051D2997ef0a62 (100 ROSE)
+(4) 0xf6FdcacbA93A428A07d27dacEf1fBF25E2C65B0F (100 ROSE)
+
+Private Keys
+==================
+(0) 0x160f52faa5c0aecfa26c793424a04d53cbf23dcad5901ce15b50c2e85b9d6ca7
+(1) 0x0ba685723b47d8e744b1b70a9bea9d4d968f60205385ae9de99865174c1af110
+(2) 0xfa990cf0c22af455d2734c879a2a844ff99bd779b400bb0e2919758d1be284b5
+(3) 0x3bf225ef73b1b56b03ceec8bb4dfb4830b662b073b312beb7e7fec3159b1bb4f
+(4) 0xad0dd7ceb896fd5f5ddc76d56e54ee6d5c2a3ffeac7714d3ef544d3d6262512c
+
+HD Wallet
+==================
+Mnemonic:       bench remain brave curve frozen verify dream margin alarm world repair innocent
+Base HD Path:   m/44'/60'/0'/0/%d
+
+WARNING: Emerald is running in ephemeral mode. The chain state will be lost after you quit!
+
+Listening on http://localhost:8545 and ws://localhost:8546
+```
+
+Those familiar with local dApp environments will find the output above similar
+to `geth --dev` or `ganache-cli` commands or the `geth-dev-assistant` npm
+package. `emerald-dev` will spin up a private Oasis network locally, generate
```

Comment:
Could you replace
``` `emerald-dev` ``` with ``` The [oasisprotocol/emerald-dev] ``` or ``` The [emerald-dev] ```

And similarly, use ``` [oasisprotocol/emerald-dev] ``` or ``` [emerald-dev] ``` throughout the document.